1-callbacks/solution.js

```diff
@@ -32,16 +32,54 @@ node solution.js name1 name2 name3
 ** give a look to node.js util.promisify, avoid to alter the validate-user.file **
 */
 
-function solution() {
-    // YOUR SOLUTION GOES HERE
-
-    // you get your 5 names here
-
-    // iterate the names array and validate them with the method
+const validateUser = require('./validate-user');
 
-    // log the final result
+function solution() {
+  // YOUR SOLUTION GOES HERE
+  // you get your 5 names here
+  const names = ['Joh', 'Mary', 'laura', 'paola', 'kenia'];
+  let sucessNames = '';
+  let failedName = '';
+
+  for (ValidateName of names) {
+    validateUser(ValidateName, (error, resp) => {
+      if (error != null) {
+        if (failedName == '') {
+          failedName = 'Failed';
+          console.log(failedName, '\n');
+        }
+
+        console.log(error);
+      } else {
+        if (sucessNames == '') {
+          sucessNames = 'Success';
+          console.log(sucessNames, '\n');
+        }
+        console.log('Id: ', resp.id);
+        console.log('Name:', resp.name);
+      }
+    });
```

Comment:
I'm thinking the question is asking for you to log every name that's not valid at the end of the function in the format `User <name> is not allowed`. This is close but you'll need to change a few things to get there

Reply:
Hey Donovan is a pleasure to meet you and i'm really excited i will do my best and learn new things from you! 

Regarding first challenge: i wanted to use the callback hell but was really frustrating, so i wanted to usea a different way to reach the answer and the final log but you are right is not valid at the end. i will takes notes about it